server/src/main/java/com/microsoft/java/bs/core/internal/services/BuildTargetService.java

```diff
@@ -345,22 +353,52 @@ public CompileResult compile(CompileParams params) {
    * clean the build targets.
    */
   public CleanCacheResult cleanCache(CleanCacheParams params) {
-    StatusCode code = runTasks(params.getTargets(), this::getCleanTaskName);
+    Function<Map<String, Set<BuildTargetIdentifier>>,
+        Function<BuildClient, ProgressReporter>> reporterCreator =
+            fullTaskPathMap -> buildClient ->
+                new DefaultProgressReporter(client);
```

Comment:
Overall the idea looks good to me. But the code here is a little bit 🤯 to me.

Can we simplify it a little bit?

Reply:
I've created an interface to make the definition clearer